### PR TITLE
Prevent eng/common PR trigger for template pipeline, close test PRs

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Prevent template pipeline run triggered by eng/common changes.
Auto close update version Pull Requests for template pipelines.

This includes some of the changes from https://github.com/Azure/azure-sdk-for-net/pull/16104 as other changes on that PR will take longer to complete.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 1]
```diff
@@ -216,6 +216,7 @@ stages:
                           PRBranchName: increment-package-version-${{ parameters.ServiceDirectory }}-$(Build.BuildId)
                           CommitMsg: "Increment package version after release of ${{ artifact.name }}"
                           PRTitle: "Increment version for ${{ parameters.ServiceDirectory }} releases"
+                          CloseAfterOpenForTesting: $(TestPipeline)
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:219]: Don't we need this for other places like the docs PR's as well?

Comment (chidozieononiwu) [reply eng/pipelines/templates/stages/archetype-net-release.yml:219]: That has been added at https://github.com/Azure/azure-sdk-for-net/blob/81e43256c59927240501e9f219134c6f5ddf263b/eng/common/pipelines/templates/steps/create-pull-request.yml#L89

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ jobs:
     steps:
       - task: PowerShell@2
         displayName: Prep template pipeline for release
-        condition: and(succeeded(),eq(variables['TestPipeline'],'true'))
+        condition: and(succeeded(),eq('${{ parameters.TestPipeline }}','true'))
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:10]: Lets switch to a template condition.

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 3]
```diff
@@ -56,7 +56,7 @@ stages:
                     - checkout: self
                     - task: PowerShell@2
                       displayName: Prep template pipeline for release
-                      condition: and(succeeded(),eq(variables['TestPipeline'],'true'))
+                      condition: and(succeeded(),eq('${{ parameters.TestPipeline }}','true'))
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:59]: We should define the TestPipeline parameter in this file and set the default to either empty or false.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 2 above)
Comment (benbp) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: @weshaggard I'm curious, what's your thinking on condition vs. template condition in general? In my mind trying to use as little templating syntax as possible makes the yaml easier to comprehend in the long run. In this case I can see how it's a little more obvious to use the template syntax as opposed to nesting logic within the `and` one-liner. Are there other things I should be considering?

**Pull request merged**